src/UCTSearch.cpp

```diff
@@ -269,16 +269,17 @@ int UCTSearch::est_playouts_left() const {
     auto playouts = m_playouts.load();
     if (m_target_time < 0) {
         // No time control, use playouts or visits.
-        const auto playouts_left = std::min(m_maxplayouts - playouts,
-                                            m_maxvisits - m_root->get_visits());
+        const auto playouts_left =
+                std::max(0, std::min(m_maxplayouts - playouts,
+                                     m_maxvisits - m_root->get_visits()));
         return playouts_left;
     } else if (elapsed_millis < 1000 || playouts < 100) {
         // Until we reach 1 second or 100 playouts playout_rate
         // is not reliable, so just return max.
         return MAXINT_DIV2;
     } else {
         const auto playout_rate = 1.0f * playouts / elapsed_millis;
-        const auto time_left = m_target_time - elapsed_millis;
+        const auto time_left = std::max(0l, m_target_time - elapsed_millis);
         return static_cast<int>(std::ceil(playout_rate * time_left));
```

Comment:
Note that in theory (although unlikely in practice) this could overflow during cast to int - which can result in a negative number even with positive input.

Reply:
Ugh, will have to think about that one. I guess it was a problem before this PR right?

Otherwise does this look good? I'm trying to resist my urge to pull this quickly so we can release v0.6  :)

Reply:
10,000NPS * 3600s*24hr = 864,000,000. Getting a little close to 2 billion max int. I bet some people will try something like this.  :(

Reply:
Yes, otherwise this looks fine.